sway/input/seat.c

```diff
@@ -453,6 +474,51 @@ void seat_set_focus_layer(struct sway_seat *seat,
 	}
 }
 
+void seat_set_exclusive_client(struct sway_seat *seat,
+		struct wl_client *client) {
+	if (!client) {
+		seat->exclusive_client = client;
+		// Triggers a refocus of the topmost surface layer if necessary
+		// TODO: Make layer surface focus per-output based on cursor position
+		for (int i = 0; i < root_container.children->length; ++i) {
+			struct sway_container *output = root_container.children->items[i];
+			if (!sway_assert(output->type == C_OUTPUT,
+						"root container has non-output child")) {
+				continue;
+			}
+			arrange_layers(output->sway_output);
+		}
+		return;
+	}
+	if (seat->focused_layer) {
+		if (wl_resource_get_client(seat->focused_layer->resource) != client) {
+			seat_set_focus_layer(seat, NULL);
+		}
+	}
+	if (seat->has_focus) {
+		struct sway_container *focus = seat_get_focus(seat);
+		if (focus->type == C_VIEW && wl_resource_get_client(
+					focus->sway_view->surface->resource) != client) {
+			seat_set_focus(seat, NULL);
+		}
+	}
+	if (seat->wlr_seat->pointer_state.focused_client) {
```

Comment:
I don't want library users to dig into `pointer_state`. What about making a getter for this now that it's important?

Reply:
+1